services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,38 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if(xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 10) != pdPASS) return ERR_CODE_INVALID_QUEUE_MSG;
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+
+  /* Get temp */
+  float temp;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp);
+
+  if(temp >= LM75BD_DEFAULT_HYST_THRESH){
+    overTemperatureDetected();
+  } else {
+    safeOperatingConditions();
+  }
+
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_t rxEvent;
+  float temp;
```

Comment:
Limit scope of these variables. They aren't used outside the while loop, so they don't need to be declared outside of it.

Reply:
Done, will push change.